src/components/mesh.tsx

```diff
@@ -0,0 +1,52 @@
+import { JSX, ReactNode, forwardRef, useRef, useState } from 'react';
+import { ThreeEvent } from '@react-three/fiber';
+import * as THREE from 'three';
+
+export type GenericMeshProps = JSX.IntrinsicElements['mesh'] & {
+  material?: ReactNode;
+  parentCallback?: (ref: React.RefObject<THREE.Mesh>) => void;
+  children?: ReactNode;
+};
+
+const GenericMesh = forwardRef<THREE.Mesh, GenericMeshProps>(
```

Comment:
`forwardRef` jest [deprecated](https://react.dev/reference/react/forwardRef). zrobilbym z tego po prostu `function`. `ref` nie musi byc podawany jako argument, bo tworzymy juz `localRef`. to na razie wystarczy